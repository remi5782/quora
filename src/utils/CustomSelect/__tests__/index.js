import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import Select from '../index';
import renderer from 'react-test-renderer';

test('Checking Select Comp with different inputs', () => {
    let checkingProps = { label: 'TagSelect', options:[{label: 'summary', value:'summary'},{label:'comment', value:'comment'}], handleSelect:()=>{} }
    let component = renderer.create(
        <BrowserRouter>
        <Select {...checkingProps}/>
        </BrowserRouter>
        
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    checkingProps = { label: 'TagSelect', options:[], handleSelect:()=>{} }
    component = renderer.create(
        <BrowserRouter>
        <Select {...checkingProps}/>
        </BrowserRouter>
        
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
});