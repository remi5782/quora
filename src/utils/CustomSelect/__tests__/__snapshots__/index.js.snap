// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  props with different values 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
    htmlFor="filled-age-native-simple"
  >
    Age
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      multiple={false}
      name="TagSelect"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    >
      <option
        value="summary"
      >
        summary
      </option>
      <option
        value="comment"
      >
        comment
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Card  props with different values 2`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
    htmlFor="filled-age-native-simple"
  >
    Age
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      className="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      multiple={false}
      name="TagSelect"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    >
      <option
        value="None"
      >
        NA
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
